debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -360,14 +409,45 @@ public JdbcConnection execute(String... sqlStatements) throws SQLException {
      * @throws SQLException if there is an error connecting to the database or executing the statements
      */
     public JdbcConnection execute(Operations operations) throws SQLException {
+        return closeOnExecute((conn) -> {
+            try (Statement statement = conn.createStatement();) {
+                operations.apply(statement);
+                if (!conn.getAutoCommit()) {
+                    conn.commit();
+                }
+            }
+        });
+    }
+
+    private JdbcConnection closeOnExecute(IExecutable executable) throws SQLException {
```

Comment:
When you have ConnectionPool, after using connection we need to close it otherwise how do you return connection to pool.

This will not really close the connection underneath instead it will return to pool and reuse it for next requests.